lib/Models/ElevationCatalogItem.js

```diff
@@ -0,0 +1,237 @@
+     * Gets or sets the result of calling the elevation function for elevation.
+     * @type {Object}
+     */
+    this.elevationResult = undefined;
+
+    /**
+     * Gets or sets the result of calling the elevation function for water table.
+     * @type {Object}
+     */
+    this.waterResult = undefined;
+
+    /**
+     * Name of chart to create
+     * @type {Object}
+     */
+    this.chartName = undefined;
+
+    CatalogItem.call(this, terria);
+}
+
+inherit(CatalogItem, ElevationCatalogItem);
+
+defineProperties(ElevationCatalogItem.prototype, {
+    /**
+     * Gets the type of data member represented by this instance.
+     * @memberOf ElevationCatalogItem.prototype
+     * @type {String}
+     */
+    type : {
+        get : function() {
+            return 'elevation-result';
+        }
+    },
+
+    /**
+     * Gets a human-readable name for this type of data source, 'Elevation Result'.
+     * @memberOf ElevationCatalogItem.prototype
+     * @type {String}
+     */
+    typeName : {
+        get : function() {
+            return 'Elevation Result';
+        }
+    },
+
+    /**
+     * Gets the data source associated with this catalog item.
+     * @memberOf ElevationCatalogItem.prototype
+     * @type {DataSource}
+     */
+    dataSource : {
+        get : function() {
+            return defined(this._geoJsonItem) ? this._geoJsonItem.dataSource : undefined;
+        }
+    }
+});
+
+ElevationCatalogItem.prototype._load = function() {
+    if (defined(this._geoJsonItem)) {
+        this._geoJsonItem._hide();
+        this._geoJsonItem._disable();
+        this._geoJsonItem = undefined;
+    }
+
+    if (!defined(this.elevationResult)) {
+        return;
+    }
+
+    var distance = this.parameters[0].getLineDistance();
+    var distanceVsElevationCsv = resultToDistanceVsElevationCsv(this.elevationResult, this.waterResult, distance);
+    var chartOptions = ' title="' + this.chartName + '" column-units="Distance (m), Meters, Meters" ';
+
+    var content = '<collapsible open="true">';
+    content += '<chart data=\'' + distanceVsElevationCsv + '\' styling="histogram"' + chartOptions + '></chart>';
+    content += '</collapsible>';
+
+    this.shortReportSections.push(new ShortReportSection({
+        content: content
+    }));
+
+    var inputsSection =
+        '<table class="cesium-infoBox-defaultTable">' +
+            '<tr>' +
+                '<td style="vertical-align: middle">line</td>' +
+                '<td style="padding-left: 4px">' + this.parameters[0].formatValueAsString() + '</td>' +
+            '</tr>' +
+            '<tr>' +
+                '<td style="vertical-align: middle">distance</td>' +
+                '<td style="padding-left: 4px">' + this.parameters[0].getLineDistance() + '</td>' +
+            '</tr>' +
+            '<tr>' +
+                '<td style="vertical-align: middle">count</td>' +
+                '<td style="padding-left: 4px">500</td>' +
+            '</tr>' +
+        '</table>';
```

Comment:
One thought that occurs to me here - could you load this as a mustache template, eg. see https://github.com/TerriaJS/terriajs/blob/master/lib/Models/SensorObservationServiceCatalogItem.js#L344-L345